File: Top Interview 150/Two Pointers/[11] Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int start = 0;
        int end = height.size() - 1;
        int maxWater = 0;

        while (start < end) {
            int h = min(height[start], height[end]);
            int currWater = h * (end - start);
            if (currWater > maxWater) maxWater = currWater;
            if (height[start] < height[end]) start++;
            else end--;
        }
        return maxWater;
    }
};

File: Leetcode 75/Two Pointers/[11] Container With Most Water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        left = 0
        right = len(height) - 1
        maxVol = 0
        while left < right:
            vol = min(height[left], height[right]) * (right - left)
            if vol > maxVol:
                maxVol = vol
            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
        return maxVol